### PR TITLE
[Merged by Bors] - Feat: port Order.Filter.Curry

### PR DESCRIPTION
**Pull request by urkud**
---

- [x] depends on: #1750
- [x] depends on: #1772

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#1750
* leanprover-community/mathlib4#1772
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#1750~~
* ~~leanprover-community/mathlib4#1772~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Order/Filter/Curry.lean [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+## Main definitions
+
+* `filter.curry`: A binary operation on filters which represents iterated limits
+
+## Main statements
+
+* `filter.eventually_curry_iff`: An alternative definition of a curried filter
+* `filter.curry_le_prod`: Something that is eventually true on the a product filter is eventually
+   true on the curried filter
```
Comment (ChrisHughes24) [inline Mathlib/Order/Filter/Curry.lean:41]:
```suggestion
* `Filter.curry`: A binary operation on filters which represents iterated limits

## Main statements

* `Filter.eventually_curry_iff`: An alternative definition of a curried filter
* `Filter.curry_le_prod`: Something that is eventually true on the a product filter is eventually
   true on the curried filter
```

File: Mathlib/Order/Filter/Curry.lean [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+uniform convergence, curried filters, product filters
+-/
+
+
+namespace Filter
+
+variable {α β γ : Type _}
+
+/-- This filter is characterized by `filter.eventually_curry_iff`:
```
Comment (ChrisHughes24) [inline Mathlib/Order/Filter/Curry.lean:53]:
```suggestion
/-- This filter is characterized by `Filter.eventually_curry_iff`:
```

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**